src/Honeycomb.Samplers/DeterministicSampler.cs

```diff
@@ -0,0 +1,72 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Security.Cryptography;
+using System.Text;
+using OpenTelemetry.Trace;
+
+namespace Honeycomb.Samplers
+{
+    public class DeterministicSampler : Sampler, IDisposable
+    {
+        private const string DescriptionFormat = "DeterministicSampler({0})";
+        private const string SampleRateAttributeName = "sampleRate";
+        private const string Hyphen = "-";
+        private const int Zero = 0;
+        private const int Four = 4;
+        private const int One = 1;

```

Comment:
it seems very odd to me to create constants for numbers, are we expecting the value of `Four` to change independently of 4?

Reply:
Not sure if it's better from an allocation perspective.  Not sure about naming them like that, but having the values you want to use as a const is a common pattern.

Reply:
I can see that if there is some semantic meaning to them like `const int maxSampleRate = 4` but the constant of four is 4! 😂 not a deal breaker for me though, just seemed odd

Reply:
Me too, but I side-stepped the naming issue :D

Reply:
I agree these are unusual but I was going for as performant as possible by using as many const's when ever I could. The .NET CLR is pretty good at inlining optimisations but this takes out the decision.

The names aren't particularly descriptive, but they are used to determine both default sample rates and index locations. I'll split them out to different fields for easier readability.